andela/neon-ah-backend
# Adds Role-based functionality

#### What does this PR do?
- Adds Role-based functionality to the App. That is, a super admin can assign users to be an admin.

#### Description of Task to be completed?
- Create Role-based functionality.
- Set up tests.

#### How should this be manually tested?
- Use `git fetch` to clone this branch.
- Switch to this branch in your machine.
- Run `npm run dev` to start the server.
- Using postman, send a `put` request to `http://localhost:3000/api/v1/users/<usernameofverifieduser>`
- If you do not have a verified user in your DB, create one by signing up.
- This request should be sent with a superAdminToken as a bearer. 
- You get a response in  format:
```
{ "status": "success",
    "data": {
        "statusCode": 200,
        "message": "The User role has been upgraded to Admin",
        "error": null,
        "payload": {
            
        }
    }
}
```
- You can test this feature by running `npm t`

#### What are the relevant pivotal tracker stories?
[Link to Pivotal Tracker Story](https://www.pivotaltracker.com/n/projects/2229851/stories/162414289)
